### PR TITLE
SWI-2779 Add Support for Custom Badges

### PR DESCRIPTION
**Pull request by ckoegel**

### CODE REVIEW

**Comment by ckoegel** [comment]
@mm-gmbd I've taken the work you did to add support for the `x-badges` field and worked it onto our current fork. This should be ready to go and just needs review from someone else on my team. Once its approved and merged, we'll do a release and it'll be available on our public package that you're free to use! If you need another custom badge type let us know and we can add it to this PR, or if its after we've merged this, feel free to open a PR to add it and we'll try to get it added asap. Thanks for getting the ball rolling on this feature 🚀

**Review by ajrice6713 — APPROVED** [review]

**Pull request merged**
